### PR TITLE
Suren Simonyan Lab4 Var2

### PR DESCRIPTION
**Pull request by SSuren4ik**
I closed the last pull request, I forgot to change the signature when copying the auxiliary function, I did it again so as not to interfere

### CODE REVIEW

File: mlir/lib/Transforms/lab4/simonyan_suren/SimonyanSurenFMAPass.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    addOp.replaceAllUsesWith(fma);
+    addOp.erase();
+  }
+
+public:
+  void runOnOperation() override {
+    ModuleOp module = getOperation();
+    module.walk([&](Operation *operation) {
+      // Add operation.
+      if (auto addOp = dyn_cast<LLVM::FAddOp>(operation)) {
```
Comment (m-ly4) [inline mlir/lib/Transforms/lab4/simonyan_suren/SimonyanSurenFMAPass.cpp]:
```suggestion
    module.walk([](LLVM::FAddOp addOp) {
```

File: mlir/lib/Transforms/lab4/simonyan_suren/SimonyanSurenFMAPass.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          handleMulOp(addOp, mulOpLHS, addRHS);
+        } else if (auto mulOpRHS = addRHS.getDefiningOp<LLVM::FMulOp>()) {
+          handleMulOp(addOp, mulOpRHS, addLHS);
+        }
+      }
+    });
+
+    module.walk([&](Operation *operation) {
+      // Mul operation.
+      if (auto mulOp = dyn_cast<LLVM::FMulOp>(operation)) {
```
Comment (m-ly4) [inline mlir/lib/Transforms/lab4/simonyan_suren/SimonyanSurenFMAPass.cpp]:
```suggestion
    module.walk([](LLVM::FMulOp mulOp) {
```

File: mlir/lib/Transforms/lab4/simonyan_suren/SimonyanSurenFMAPass.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Tools/Plugins/PassPlugin.h"
+
+using namespace mlir;
+
+namespace {
+class SimonyanSurenFMAPass
+    : public PassWrapper<SimonyanSurenFMAPass, OperationPass<ModuleOp>> {
```
Comment (m-ly4) [inline mlir/lib/Transforms/lab4/simonyan_suren/SimonyanSurenFMAPass.cpp]: Please use narrower operation type for scheduling the pass on.

**Review by m-ly4 — CHANGES_REQUESTED** [review]

File: mlir/lib/Transforms/lab4/simonyan_suren/SimonyanSurenFMAPass.cpp (hunk 3 above)
Comment (SSuren4ik) [reply mlir/lib/Transforms/lab4/simonyan_suren/SimonyanSurenFMAPass.cpp]: Сhanges have been made

**Review by aobolensk — APPROVED** [review]

**Review by m-ly4 — APPROVED** [review]

**Comment by m-ly4** [comment]
@SSuren4ik Could you please re-open your MR and fix tests?

**Pull request merged**